components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIProductMappingUtil.java

```diff
@@ -0,0 +1,313 @@
+        }
+
+        if (model.getVisibleTenants() != null) {
+            dto.setVisibleRoles(Arrays.asList(model.getVisibleTenants().split(",")));
+        }
+
+        APIBusinessInformationDTO apiBusinessInformationDTO = new APIBusinessInformationDTO();
+        apiBusinessInformationDTO.setBusinessOwner(model.getBusinessOwner());
+        apiBusinessInformationDTO.setBusinessOwnerEmail(model.getBusinessOwnerEmail());
+        apiBusinessInformationDTO.setTechnicalOwner(model.getTechnicalOwner());
+        apiBusinessInformationDTO.setTechnicalOwnerEmail(model.getTechnicalOwnerEmail());
+        dto.setBusinessInformation(apiBusinessInformationDTO);
+
+        return dto;
+    }
+
+    private static APIProductStatus mapStatusFromDTOToAPIProduct(String status) {
+        // switch case statements are not working as APIStatus.<STATUS>.toString() or APIStatus.<STATUS>.getStatus()
+        //  is not a constant
+        if (status.equals(APIProductStatus.BLOCKED.toString())) {
+            return APIProductStatus.BLOCKED;
+        } else if (status.equals(APIProductStatus.CREATED.toString())) {
+            return APIProductStatus.CREATED;
+        } else if (status.equals(APIProductStatus.PUBLISHED.toString())) {
+            return APIProductStatus.PUBLISHED;
+        } else if (status.equals(APIProductStatus.DEPRECATED.toString())) {
+            return APIProductStatus.DEPRECATED;
+        } else {
+            return null; // how to handle this?
+        }
+    }
+
+    private static String mapSubscriptionAvailabilityFromDTOtoAPIProduct(
+            APIProductDTO.SubscriptionAvailabilityEnum subscriptionAvailability) {
+        switch (subscriptionAvailability) {
+            case current_tenant:
+                return APIConstants.SUBSCRIPTION_TO_CURRENT_TENANT;
+            case all_tenants:
+                return APIConstants.SUBSCRIPTION_TO_ALL_TENANTS;
+            case specific_tenants:
+                return APIConstants.SUBSCRIPTION_TO_SPECIFIC_TENANTS;
+            default:
+                return null; // how to handle this? 500 or 400
+        }
+
+    }
+
+    private static APIProductDTO.SubscriptionAvailabilityEnum mapSubscriptionAvailabilityFromAPIProductToDTO(
+            String subscriptionAvailability) {
+
+        switch (subscriptionAvailability) {
+            case APIConstants.SUBSCRIPTION_TO_CURRENT_TENANT :
+                return APIProductDTO.SubscriptionAvailabilityEnum.current_tenant;
+            case APIConstants.SUBSCRIPTION_TO_ALL_TENANTS :
+                return APIProductDTO.SubscriptionAvailabilityEnum.all_tenants;
+            case APIConstants.SUBSCRIPTION_TO_SPECIFIC_TENANTS :
+                return APIProductDTO.SubscriptionAvailabilityEnum.specific_tenants;
+            default:
+                return null; // how to handle this?
+        }
+
+    }
+
+    private static String mapVisibilityFromDTOtoAPIProduct(APIProductDTO.VisibilityEnum visibility) {
```

Comment:
the functionality if these two methods looks similar, can't we can't we have single method